webui/react/src/hooks/useModal/UserSettings/useModalCreateUser.tsx

```diff
@@ -33,6 +41,7 @@ export const ROLE_NAME = 'roles';
 interface Props {
   form: FormInstance;
   groups: V1GroupSearchResult[];
+  roles: UserRole[] | null;
```

Comment:
i might miss something, but it can be `roles: UserRole[]`, right?
we can represent empty list as null.

Reply:
@keita-determined `null` is the loading state.